docs/docs/core/permissions/how-they-work.mdx

```diff
@@ -0,0 +1,150 @@
+---
+title: How Permissions Work
+tags:
+  - permissions
+  - RBAC
+---
+
+## Overview
+
+The RBAC permission system relies under the hood on the [CASL](https://casl.js.org) library.
+
+
+A `permission` is the combination of an `action`, a `subject`, some `properties` and some `conditions`.
+The logic is that: a user can perform an `action` on a `subject` and some of its `properties` only if it its role has the `permission` and if the `conditions` associated to the permission pass.
+The creation and edition of permissions is done through the [edit page](http://localhost:1337/admin/settings/roles/2) of a role by checking or unchecking checkboxes.
+
+Example 1: the action `update` can be applied on the subject `article` and property `title`, this result is the permission to update the title of an article.
+
+Example 2: the condition `isCreator` added to the previous permission will ensure that only the creator of the article can update the title.
+
+Example 3: the action `access the Marketplace page` has no subject to apply on. The permission only contains the action `admin::marketplace.read`.
+
+
+The permissions are attached to a role. Each role can have different permissions independentely. There is no inheritance system.
+To check if an action can be performed by a user, frontend and backend simply retrieve the list of the permissions associated with the user's roles and check if it contains the permission associated with the action about to be performed. The backend also runs the associated conditions.
+
+:::tip
+If a user has several role, the user will be allowed to perform an action if at least one of its roles is allowed to perform it.
+:::
+
+## Actions
+
+### Action definition
+
+An action contains the following information:
+
+| key                           | description                                                                                                                                       | type   | required | Default value | example                    |
+| ----------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------- | ------ | -------- | ------------- | -------------------------- |
+| **uid**                       | An id that has to be unique within the plugin.                                                                                                    | string | required | -             | `'audit-logs.read'`        |
```

Comment:
On what do we base this uid?

Something like <nameoffeature>.<action> ?

Reply:
It depends, it can be the name of the entity. It's nice to follow how it was done before but it's not strictly consistent. But ultimately the only thing that matter is that it is unique.
You can find a list of existing ids here: https://docs.google.com/spreadsheets/d/13kIEKshstD9WJqWBctxb8TeAIHDZCw5ttg7vgzUAsBU/edit#gid=0